### Auctions data

The auction dataset has been taken from this <a href="https://github.com/jasonshi10/art_auction_valuation">repository</a> which has taken data webscraping the old website of Blouin Art Sales index. During 2014 that website made available a lot of auction data for free <a href="https://web.archive.org/web/20141018210450/http://artsalesindex.artinfo.com/asi/search/artistLanding.ai">(see here)</a>, now data is available only under subscription as other websites like Artsy. 


In [7]:
import pandas as pd
pd.set_option('display.max_columns', None)
auctions = pd.read_csv("auctions.csv", sep=";", encoding="iso-8859-1")
auctions

Column1             artist   country yearOfBirth  yearOfDeath  \
0          0.0            Mario A     Swiss        1959          NaN   
1          1.0            Mario A     Swiss        1959          NaN   
2          2.0         A E Cremer    French       20thC          NaN   
3          3.0  A G Schultz & Co.  American    est1890s          NaN   
4          4.0  A G Schultz & Co.  American    est1890s          NaN   
...        ...                ...       ...         ...          ...   
44097  41290.0        Mark Rothko  American      1903.0      19700.0   
44098  41291.0        Mark Rothko  American      1903.0      19700.0   
44099  41292.0        Mark Rothko  American      1903.0      19700.0   
44100  41293.0        Mark Rothko  American      1903.0      19700.0   
44101  41294.0        Mark Rothko  American      1903.0      19700.0   

                                                    name  year       price  \
0                              The world is beautyful #5  2004     53150.0   
1                             The world is beautyful #13  2004     73830.0   
2                                            Spot Lights   NaN     20900.0   
3                                Sugar/Sweetmeat Baskets   NaN      6150.0   
4      An American Silver Five-piece Tea And Coffee S...  1908     81250.0   
...                                                  ...   ...         ...   
44097                                           Untitled  1961  17000000.0   
44098                                           Untitled  1968   3000000.0   
44099                                           Untitled  1959   2800000.0   
44100                        Green, White, Green on Blue  1969   3229380.0   
44101                                         Red on red  1968   3300000.0   

                            material height  width  \
0           laserchrome_print_diasec  29.53  39.37   
1         laserchrom_print_(diasec.)  29.53  39.37   
2                black-painted_metal    NaN    NaN   
3                    Sterling_Silver    6.5   5.75   
4                             silver    NaN    NaN   
...                              ...    ...    ...   
44097                  oil_on_canvas   93.0   80.0   
44098     acrylic_on_paper_on_canvas   24.0   18.0   
44099  oil_on_paper_laid_on_masonite   38.0   25.0   
44100          oil_on_paper_on_board   71.0   41.0   
44101         oil_on_paper_on_canvas   40.0   25.0   

                                                    link  \
0      http://artsalesindex.artinfo.com/asi/lots/3456055   
1      http://artsalesindex.artinfo.com/asi/lots/3456059   
2      http://artsalesindex.artinfo.com/asi/lots/5250074   
3      http://artsalesindex.artinfo.com/asi/lots/5269624   
4      http://artsalesindex.artinfo.com/asi/lots/3900522   
...                                                  ...   
44097  http://artsalesindex.artinfo.com/asi/lots/2495060   
44098  http://artsalesindex.artinfo.com/asi/lots/2495081   
44099  http://artsalesindex.artinfo.com/asi/lots/2495098   
44100  http://artsalesindex.artinfo.com/asi/lots/2251703   
44101  http://artsalesindex.artinfo.com/asi/lots/2085679   

                                                  source dominantColor  \
0      http://artinfo-images-350.s3.amazonaws.com/asi...       yellows   
1      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
2      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
3      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
4      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
...                                                  ...           ...   
44097  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44098  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44099  http://artinfo-images-350.s3.amazonaws.com/177...        blacks   
44100  http://artinfo-images-350.s3.amazonaws.com/S11...        blacks   
44101  http://artinfo

In [8]:
for i, item in enumerate(auctions['artist']):
    if isinstance(item, str) and 'After  ' in item:
        auctions.at[i, 'artist'] = item.replace('After  ', '')
auctions
#modificato oggi 

Column1             artist   country yearOfBirth  yearOfDeath  \
0          0.0            Mario A     Swiss        1959          NaN   
1          1.0            Mario A     Swiss        1959          NaN   
2          2.0         A E Cremer    French       20thC          NaN   
3          3.0  A G Schultz & Co.  American    est1890s          NaN   
4          4.0  A G Schultz & Co.  American    est1890s          NaN   
...        ...                ...       ...         ...          ...   
44097  41290.0        Mark Rothko  American      1903.0      19700.0   
44098  41291.0        Mark Rothko  American      1903.0      19700.0   
44099  41292.0        Mark Rothko  American      1903.0      19700.0   
44100  41293.0        Mark Rothko  American      1903.0      19700.0   
44101  41294.0        Mark Rothko  American      1903.0      19700.0   

                                                    name  year       price  \
0                              The world is beautyful #5  2004     53150.0   
1                             The world is beautyful #13  2004     73830.0   
2                                            Spot Lights   NaN     20900.0   
3                                Sugar/Sweetmeat Baskets   NaN      6150.0   
4      An American Silver Five-piece Tea And Coffee S...  1908     81250.0   
...                                                  ...   ...         ...   
44097                                           Untitled  1961  17000000.0   
44098                                           Untitled  1968   3000000.0   
44099                                           Untitled  1959   2800000.0   
44100                        Green, White, Green on Blue  1969   3229380.0   
44101                                         Red on red  1968   3300000.0   

                            material height  width  \
0           laserchrome_print_diasec  29.53  39.37   
1         laserchrom_print_(diasec.)  29.53  39.37   
2                black-painted_metal    NaN    NaN   
3                    Sterling_Silver    6.5   5.75   
4                             silver    NaN    NaN   
...                              ...    ...    ...   
44097                  oil_on_canvas   93.0   80.0   
44098     acrylic_on_paper_on_canvas   24.0   18.0   
44099  oil_on_paper_laid_on_masonite   38.0   25.0   
44100          oil_on_paper_on_board   71.0   41.0   
44101         oil_on_paper_on_canvas   40.0   25.0   

                                                    link  \
0      http://artsalesindex.artinfo.com/asi/lots/3456055   
1      http://artsalesindex.artinfo.com/asi/lots/3456059   
2      http://artsalesindex.artinfo.com/asi/lots/5250074   
3      http://artsalesindex.artinfo.com/asi/lots/5269624   
4      http://artsalesindex.artinfo.com/asi/lots/3900522   
...                                                  ...   
44097  http://artsalesindex.artinfo.com/asi/lots/2495060   
44098  http://artsalesindex.artinfo.com/asi/lots/2495081   
44099  http://artsalesindex.artinfo.com/asi/lots/2495098   
44100  http://artsalesindex.artinfo.com/asi/lots/2251703   
44101  http://artsalesindex.artinfo.com/asi/lots/2085679   

                                                  source dominantColor  \
0      http://artinfo-images-350.s3.amazonaws.com/asi...       yellows   
1      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
2      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
3      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
4      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
...                                                  ...           ...   
44097  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44098  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44099  http://artinfo-images-350.s3.amazonaws.com/177...        blacks   
44100  http://artinfo-images-350.s3.amazonaws.com/S11...        blacks   
44101  http://artinfo

In [9]:
from collections import Counter
liss = {}
for column_name in auctions.columns:
    if column_name == "artist":
        for value in auctions[column_name]:
            if value not in liss:
                liss[value] = 1
            else:
                liss[value] += 1
k = Counter(liss)
 
# Finding 3 highest values
high = k.most_common(4) 
 
print("Initial Dictionary:")
print(liss, "\n")
 
 
print("Dictionary with 3 highest values:")
print("Keys: Values")
 
for i in high:
    print(i[0]," :",i[1]," ")

Initial Dictionary:
{'Mario A': 2, 'A E Cremer': 1, 'A G Schultz & Co.': 2, 'A H': 1, 'A H G': 1, 'A K': 1, 'A Kassen': 1, 'A Lao': 4, 'A M': 1, 'A Sun Wu': 4, "Maria A'Becket": 4, "A'erbai": 1, 'A-One': 2, 'A.CE': 1, 'A1one': 6, 'Dirk van der Aa': 6, 'Jorgen Aabye': 5, 'Christian Aabye-Talge': 1, 'Attributed to  Hans von Aachen': 2, 'Circle of  Hans von Aachen': 2, 'Studio of  Hans von Aachen': 1, 'Hans von Aachen': 2, 'Peder Pedersen Aadnes': 1, 'Arvid Aae': 2, 'Carl Frederic Aagaard': 4, 'Martin Aagaard': 2, 'H P Aaland': 1, 'Alvar Aalto': 11, nan: 2837, 'Ilmari Aalto': 4, 'Veikko Aaltona': 5, 'Aarre Aaltonen': 2, 'Waino Aaltonen': 7, 'Martin Aalund': 3, 'Asbjorn Aamodt': 2, 'Susanne Aamund': 1, 'Jos Aanraad': 3, 'Leny Aardse': 1, 'Eero Aarnio': 7, 'Jessie Aaron': 9, 'John Aaron': 1, 'Joseph Aaron': 5, 'Michel Aaron': 1, 'George Aarons': 4, 'Slim Aarons': 2, 'Carl Aarsleff': 2, 'Hans Aarsman': 1, 'Nicolaas Aartmann': 4, 'Pieter Jan Aarts': 1, 'Alf Jorgen Aas': 1, 'Nils Aas': 3, 'Wal

In [10]:
liss = []
for column_name in auctions.columns:
    if column_name == "artist":
        for value in auctions[column_name]:
            if value == "Willem de Kooning":
                liss.append(value)
print(len(liss))

381


Get only results related to sales on De Kooning works. Artworks that do not possess data about price or soldtime have been deleted from the dataset.
The dataset does not possess any data related to sellers and buyers. 
In this example we notice that 307 works over a total of 381 possess data about price and time of the transaction. 

In [11]:
import pandas as pd

new_dataframe = auctions[["artist", "name", "price", "soldtime"]].copy()
new_dataframe = new_dataframe[new_dataframe["artist"] == "Willem de Kooning"].copy()

new_dataframe.reset_index(drop=True, inplace=True)
def remove_last_zero(x):
    if str(x).endswith('0'):
        return str(x)[:-3]
    else:
        return x

# Apply the function to the 'price' column
new_dataframe['price'] = new_dataframe['price'].apply(remove_last_zero)

for idx, row in new_dataframe.iterrows():
    if pd.isna(row["price"]) and pd.isna(row["soldtime"]):
        new_dataframe.drop(idx, inplace=True)
        
dayy = []
for i in new_dataframe['soldtime']:
    string_representation = str(i)
    split_parts = string_representation.split('-')
    day = split_parts[0]
    dayy.append(day)
new_dataframe['soldtime'] = dayy
        

new_dataframe.reset_index(drop=True, inplace=True)
new_dataframe.to_csv('auctions_data_DK.csv', index=False)
new_dataframe

artist                name     price soldtime
0    Willem de Kooning            Untitled    137000     2014
1    Willem de Kooning            Untitled   2045000     2014
2    Willem de Kooning            Untitled    112500     2014
3    Willem de Kooning      Untitled XVIII   4869000     2014
4    Willem de Kooning       Untitled XXIV  17525000     2014
..                 ...                 ...       ...      ...
302  Willem de Kooning             Woman V    313000     2008
303  Willem de Kooning         Untitled IV  12081000     2008
304  Willem de Kooning           With love      4400     2008
305  Willem de Kooning  Untitled (Spoleto)    121000     2008
306  Willem de Kooning            Untitled     40000     2008

[307 rows x 4 columns]